VRDR.Messaging/CodingResponseMessage.cs

```diff
@@ -140,14 +140,267 @@ public enum RaceCode
+            }
+            set
+            {
+                Record.Remove("int_rej");
+                if (value != null)
+                {
+                    if (value != 9 || value > 5 || value < 1) {
+                        throw new ArgumentException("Intentional Reject must equal 1-5, 9, or null.");
+                    }
+                    Record.Add("int_rej", new UnsignedInt((int)value));
+                }
+            }
+        }
+
+        /// <summary>ACME System Reject Enum</summary>
+        public enum ACMESystemRejectEnum
+        {
+            /// <summary>MICAR Reject - Dictionary Match</summary>
+            MICARRejectDictionaryMatch,
+            /// <summary>ACME Reject</summary>
+            ACMEReject,
+            /// <summary>MICAR Reject - Rule Application</summary>
+            MICARRejectRuleApplication,
+            /// <summary>Not Rejected</summary>
+            NotRejected,
+        }
+
+        /// <summary>ACME system reject codes (or null)</summary>
+        public ACMESystemRejectEnum? ACMESystemRejectCodes
+        {
+            get
+            {
+                string value = Record.GetSingleValue<FhirString>("sys_rej")?.Value;
+                if(value == null)
+                {
+                    return null;
+                }
+                else
+                {
+                    Enum.TryParse<ACMESystemRejectEnum>(value, out ACMESystemRejectEnum code);
+                    return code;
+                }
+            }
+            set
+            {
+                Record.Remove("sys_rej");
+                if(value != null)
+                {
+                    Record.Add("sys_rej", new FhirString(value.ToString()));
+                }
+            }
+        }
+
+        /// <summary>Place of Injury Enum</summary>
+        public enum PlaceOfInjuryEnum
+        {
+            /// <summary>Home</summary>
+            Home,
+            /// <summary>Residential Institution</summary>
+            ResidentialInstution,
+            /// <summary>School, Other Institutions, Public Administrative Area</summary>
+            SchoolOtherInstutionOrPublicAdministrativeArea,
+            /// <summary>Sports and Atheletics Area</summary>
+            SportsAndAthleticsArea,
+            /// <summary>Street/Highway</summary>
+            StreetOrHighway,
+            /// <summary>Trade and Service Area</summary>
+            TradeAndServiceArea,
+            /// <summary>Industrial and Construction Area</summary>
+            IndustrialAndConstructionArea,
+            /// <summary>Farm</summary>
+            Farm,
+            /// <summary>Other Specified Place</summary>
+            OtherSpecifiedPlace,
+            /// <summary>Unspecified Place</summary>
+            UnspecifiedPlace
+        }
+
+        /// <summary>Place of Injury (or null)</summary>
+        public PlaceOfInjuryEnum? PlaceOfInjury
```

Comment:
Question: Does this need to be a `Dictionary<PlaceOfInjuryEnum?, string>` in order to properly support values such as `OtherSpecifiedPlace`?

Reply:
I would probably go with two properties, one for the enum and a second optional one for the "specified place".